src/SimpleTracer.php

```diff
@@ -0,0 +1,92 @@
+<?php namespace JaegerPhp;
+
+use OpenTracing\Carriers\TextMap;
+use OpenTracing\Propagator;
+use OpenTracing\Reference;
+use JaegerPhp\Sampler\ProbabilisticSampler;
+use JaegerPhp\Sampler\ConstSampler;
+
+/**
+ * Create an interface to reduce instrumentation lines of code as much as possible.
+ * Usage:
+ *
+ *  $tracer = new SimpleTracer(
+ *      'server.com',
+ *      'root span name',
+ *      '0.0.0.0:5775',
+ *      [],
+ *      TRUE
+ *  );
+ *
+ *  $span1 = $tracer->createSpan('span1');
+ *  $childOfSpan1 = $tracer->createSpan('childOfSpan1', [...tags...], $span1);
+ *  ...
+ *  $tracer->finishSpan($childOfSpan1, [...tags..]);
+ *  ...
+ *  $tracer->finishSpan($span1, [...tags..]);
+ */
+
+class SimpleTracer {
+
+    public function __construct($traceLabel, $traceSpanLabel, $agentUri, $injectTarget = [], $debugTrace = FALSE, $environment = NULL) {
+        if (!$agentUri) {
+            throw new \InvalidArgumentException('$agentUri cannot be null');
+        }
+
+        try {
+            if ($environment) {
+                $_SERVER['JAEGER_TAGS'] = 'app.environment='.$environment;
+            }
+
+            $traceConfig = Config::getInstance();
+
+            if ($debugTrace) {
+                $sampler = new ConstSampler();
+            } else {
+                $sampler = new ProbabilisticSampler();
+            }
+
+            $traceConfig->setSampler($sampler);
+            $this->tracer = $traceConfig->initTrace($traceLabel, $agentUri);
+            $textMap = TextMap::create($injectTarget);
+            $spanContext = $this->tracer->extract(Propagator::TEXT_MAP, $textMap);
+            $span = $this->tracer->startSpan($traceSpanLabel, Reference::create(Reference::CHILD_OF, $spanContext));
+
+            register_shutdown_function(function() use ($traceConfig, $span) {
```

Comment:
wondering if we should separate it into two .. so that if span fails, we still finish trace, rather than orphan both of them if span throws an exception?